### ISSUE TITLE
Forward rendering seems broken

### ISSUE THREAD

**Issue by JMS55** [bevyengine/bevy#16679]
## Bevy version
72f096c91ea1056c41f2da821d310765c4482c00

## Relevant system information

```
2024-12-06T04:20:17.216608Z  INFO bevy_diagnostic::system_information_diagnostics_plugin::internal: SystemInfo { os: "Windows 11 Home", kernel: "22631", cpu: "AMD Ryzen 7 5800X 8-Core Processor", core_count: "8", memory: "15.9 GiB" }
2024-12-06T04:20:17.720519Z  INFO bevy_render::renderer: AdapterInfo { name: "NVIDIA GeForce RTX 3080", vendor: 4318, device: 8710, device_type: DiscreteGpu, driver: "NVIDIA", driver_info: "560.94", backend: Vulkan }
```

## What you did

Run the deferred_rendering example

## What went wrong

Forward rendering seems broken (forward + prepass is fine)

* POM seems broken on the cube
* Normal maps seem to be missing from the flight helmet

**Comment by pcwalton** [comment]
Bisecting this. The lack of POM on the cube repros even as far back as Sep 13. So that at least isn't a recent regression.

**Comment by DGriffin91** [comment]
Lack of parallax on the cube was also in the initial creation of this scene. Forward + prepass has never supported parallax mapping.

**Comment by JMS55** [comment]
Yes, but it's also broken in pure forward (there's some weird ghosting 🙃)

**Comment by pcwalton** [comment]
The culprit for the loss of normal mapping is the retained render world, #15320.

Note that deferred was broken at the time that commit landed (due to PCSS) and wasn't fixed until #15656, which is ddd4b4daf8bb3df1dac0fd5f453269ff4bd4f99a. So you'll want to `git cherry-pick ddd4b4daf8bb3df1dac0fd5f453269ff4bd4f99a` if you check out that commit.

**Comment by hukasu** [comment]
the issue being described here is the shadow being retained when changing from Deferred or Forward + Prepass into Forward, right?

![Image](https://github.com/user-attachments/assets/08c4461b-2018-4f37-96ce-c08e2eed4cd5)

taken using 85eceb022da0326b47ac2b0d9202c9c9f01835bb

**Comment by hukasu** [comment]
things got worse on 7aeb1c51a6e6e8ec6d0ff659f90a38e848884b94

Deferred  
![Image](https://github.com/user-attachments/assets/e54e969d-8cf7-410a-9e4d-42a3af6fd875)

Deferred -> Forward  
![Image](https://github.com/user-attachments/assets/504aaac5-7d05-4625-bbb9-9ac11d532d4f)

Forward -> Deferred  
![Image](https://github.com/user-attachments/assets/91cd3c7d-389f-44cb-b35b-bf7f9117ca80)

Deferred -> Forward + Prepass
![Image](https://github.com/user-attachments/assets/a0c8ee9d-5e71-48db-a555-da2be597282e)

Forward + Prepass -> Forward
![Image](https://github.com/user-attachments/assets/e56b3abd-2dbd-4940-ab84-443b94818776)

Forward -> Deferred
![Image](https://github.com/user-attachments/assets/d0e16306-dc91-4236-a541-0a7a13c0d710)

**Comment by hukasu** [comment]
git bisect done
`dda97880c48ff39c3a77ff2eb398c34b382cb771 is the first bad commit`

**Comment by pcwalton** [comment]
The root of the problem is that we add prepasses to the render world views but don't remove them. #17565 fixes that but can't be merged yet because it exposes other bugs.
